src/Namespace.sol

```diff
@@ -0,0 +1,578 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.14;
+
+import {ERC721} from "../lib/solmate/src/tokens/ERC721.sol";
+
+import {Owned} from "../lib/solmate/src/auth/Owned.sol";
+
+import {FixedPointMathLib} from "../lib/solmate/src/utils/FixedPointMathLib.sol";
+
+error InsufficientFunds(); // The transaction does not have enough money to pay for this.
+error Unauthorized(); // The caller is not authorized to perform this action.
+
+error InvalidCommit(); // The commitment hash was not found
+error InvalidName(); // The username had invalid characters
+error InvalidTime(); // Time is too far in the future or past
+error IncorrectOwner(); // The username is not owned by the expected address
```

Comment:
When would we use this over `Unauthorized`, line 11 ? Opportunity to simplify and keep only one?

Reply:
so the case here is that `bob` is renewing the name `@foo` that `alice` currently owns. bob wants to specify that is paying the fee for foo and for alice owning foo. 

if somehow alice sells the name to charlie or an admin reclaimed this name, bob doesn't want this transaction to complete. so if the name did change hands and it is now owned by charlie, we throw IncorrectOwner. 

Unauthorized is slightly misleading here - bob *is* authorized to pay to renew charlie's username `!@foo` its just that bob only wanted to pay it when alice owned it.

Reply:
open to other terms, I agree that incorrect owner is not immediately obvious to the reader